### PR TITLE
Adds the M579 Aerial-Defence System Carrier (AKA, the AA-APC, or LAV-AD of the future)

### PR DESCRIPTION
**Pull request by BonniePandora**
# About the pull request

Adds a new variant of the APC to the game for GM's to utilise as they see fit, and also offers a flavourful way for marines to down airborne threats such as enemy CAS (via GM intervention and such)

Very much a 'glass cannon' kind of vehicle, capable of dealing a lot of hurt but can't take much punishment in return. Adjustments to damage/range for the weapons is more than doable

Sprites done by Boom on discord largely, with some extrapolated by myself from provided sprites.

# Explain why it's good for the game

More things = gooder
Drew inspiration from the tech manual's listed variant of the basic APC, thought it would be neat to add, was a fun code project to work on.
Gives GM's more options for support assets to use as and when they want to too.

# Testing Photographs and Procedure
Tested each step of the way on a local server to ensure it worked, no issues besides placeholder sprites for much of the hardpoints.
<details>
<summary>Screenshots & Videos</summary>

image 1

image 1

</details>

# Changelog

:cl:
add: added the M579 ADS-C vehicle
add: added the new weapons, ammo, ammo-storage and viewport console for the above
fix: fixed a typo in the vehicles.dm file whilst I was at it
imageadd: new turf icon added for ADS-C interior
mapadd: added new interior for the M579
/:cl:

### CODE REVIEW

File: icons/obj/vehicles/movieapc.dmi
Comment (Doubleumc) [inline icons/obj/vehicles/movieapc.dmi:1]: Since this is a subtype of the apc, not the movie apc, it should be in `icons/obj/vehicles/apc.dmi` instead.

**Review by Doubleumc — CHANGES_REQUESTED** [review]
Code good. Just want to see the hardpoint sprites moved to the appropriate file and for it to be tested in at least one operation.

**Comment by BonniePandora** [comment]
Swapping to a draft until I can get the finished sprites, will address the merge-conflicts & requested changes when that's happened.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: icons/obj/vehicles/movieapc.dmi
Comment (BonniePandora) [reply icons/obj/vehicles/movieapc.dmi:1]: This has been addressed, made the AA APC a sub-type of the movielike one instead

File: maps/interiors/apc_aa.dmm
[the remark this answers is not in the document]
Comment (BonniePandora) [reply maps/interiors/apc_aa.dmm:1]: Provided the remaining checks pass, these changes should be in now.

**Review by Doubleumc — APPROVED** [review]

**Comment by BonniePandora** [comment]
Sprites are finished at last, ready for a possible TM & review.

File: code/game/objects/structures/gun_rack.dm [hunk 1]
```diff
@@ -85,6 +85,22 @@
 	initial_stored = 0
 	allowed_type = /obj/item/ammo_magazine/hardpoint/boyars_dualcannon
 
+/obj/structure/gun_rack/apc/frontal/aa
+	name = "vertical-launch-system ammo storage compartment"
+	icon_state = "boyars"
+	desc = "A small compartment that stores ammunition for the vehicle's VLS-array ."
```
Comment (AndroBetel) [inline code/game/objects/structures/gun_rack.dm:91]:
```suggestion
	desc = "A small compartment that stores ammunition for the vehicle's VLS-array."
```

File: code/modules/vehicles/apc/apc_aa.dm [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+		/obj/item/hardpoint/primary/aa_quadcannon,
+		/obj/item/hardpoint/secondary/towlauncher/aa,
+		/obj/item/hardpoint/support/flare_launcher,
+		/obj/item/hardpoint/locomotion/apc_wheels,
+	)
+
+	passengers_slots = 10
+
+
+/obj/vehicle/multitile/apc/movie/aa/add_seated_verbs(mob/living/M, seat)
```
Comment (AndroBetel) [inline code/modules/vehicles/apc/apc_aa.dm:17]: inherited from movie apc, not needed

File: code/modules/vehicles/apc/apc_aa.dm [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+			/obj/vehicle/multitile/proc/activate_horn,
+		))
+	else if(seat == VEHICLE_GUNNER)
+		add_verb(M.client, list(
+			/obj/vehicle/multitile/proc/switch_hardpoint,
+			/obj/vehicle/multitile/proc/cycle_hardpoint,
+			/obj/vehicle/multitile/proc/toggle_shift_click,
+		))
+
+/obj/vehicle/multitile/apc/movie/aa/remove_seated_verbs(mob/living/M, seat)
```
Comment (AndroBetel) [inline code/modules/vehicles/apc/apc_aa.dm:40]: ditto

File: code/modules/vehicles/apc/apc_aa.dm [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+	if(change_tag)
+		camera.c_tag = "#[rand(1,100)] M579 \"[nickname]\" ADS-C"
+		if(camera_int)
+			camera_int.c_tag = camera.c_tag + " interior"
+	else
+		camera.c_tag = "#[rand(1,100)] M579 ADS-C"
+		if(camera_int)
+			camera_int.c_tag = camera.c_tag + " interior"
+
+/obj/vehicle/multitile/apc/movie/aa/set_muzzle_offsets(obj/item/hardpoint/HP)
```
Comment (AndroBetel) [inline code/modules/vehicles/apc/apc_aa.dm:76]: ditto? should check if these are correct

**Review by AndroBetel — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/gun_rack.dm (hunk 1 above)
Comment (BonniePandora) [reply code/game/objects/structures/gun_rack.dm]: Fixed

File: code/modules/vehicles/apc/apc_aa.dm (hunk 2 above)
Comment (BonniePandora) [reply code/modules/vehicles/apc/apc_aa.dm]: Removed from the file

File: code/modules/vehicles/apc/apc_aa.dm (hunk 3 above)
Comment (BonniePandora) [reply code/modules/vehicles/apc/apc_aa.dm]: As above

File: code/modules/vehicles/apc/apc_aa.dm (hunk 4 above)
Comment (BonniePandora) [reply code/modules/vehicles/apc/apc_aa.dm:28]: Checked as of latest commit, values adjusted where needed

File: code/modules/vehicles/hardpoints/primary/aa_cannon.dm [hunk 5]
```diff
@@ -0,0 +1,37 @@
+/obj/item/hardpoint/primary/aa_quadcannon
+	name = "Bleihagel RE-RE965 Aerial-Defence System"
+	desc = "A compact turret with four 20mm rotary cannons mounted in it. Normally slaved to the vehicles tracking systems, this one seems to be lacking the electronics required for such."
+	icon = 'icons/obj/vehicles/hardpoints/apc.dmi'
+
+	icon_state = "quad_cannon"
```
Comment (AndroBetel) [inline code/modules/vehicles/hardpoints/primary/aa_cannon.dm:6]: image
fucked up shadow

File: code/modules/vehicles/apc/apc_aa.dm [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+				"8" = list(-56, 12)
+			)
+		if(HDPT_SUPPORT) //flare
+			HP.muzzle_flash_pos = list(
+				"1" = list(22, -20),
+				"2" = list(11, 32),
+				"4" = list(-5, 10),
+				"8" = list(37, 24)
+			)
+	return ..()
```
Comment (AndroBetel) [inline code/modules/vehicles/apc/apc_aa.dm:52]: remove this return, fucks up muzzle_flash_pos

File: code/modules/vehicles/hardpoints/secondary/aa_launcher.dm [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+	muzzle_flash_pos = list(
+		"1" = list(8, -1),
+		"2" = list(-8, -16),
+		"4" = list(5, -8),
+		"8" = list(-5, 10)
+	)
+
+	scatter = 1
+	fire_delay = 7.0 SECONDS
```
Comment (AndroBetel) [inline code/modules/vehicles/hardpoints/secondary/aa_launcher.dm:32]: feels anemic i think, maybe make it almost instant and also up max_clips a little? just a suggestion

File: code/modules/vehicles/interior/interactable/viewports.dm [hunk 8]
```diff
@@ -34,3 +34,9 @@
 	icon = 'icons/obj/vehicles/interiors/van.dmi'
 	icon_state = "windshield_viewport_top"
 	alpha = 80
+
+//anti-air APC's interior console
+/obj/structure/interior_viewport/terminal
```
Comment (AndroBetel) [inline code/modules/vehicles/interior/interactable/viewports.dm:39]: doesnt work

File: maps/interiors/apc_aa.dmm [hunk 9]
```diff
@@ -0,0 +1,482 @@
...
+/obj/structure/interior_wall/apc{
+	icon_state = "wheel_front_top_2";
+	pixel_x = -4;
+	pixel_y = -5;
+	layer = 2.01
+	},
+/turf/open/void/vehicle,
+/area/space)
+"c" = (
+/obj/structure/prop/vehicle/sensor_equipment{
```
Comment (AndroBetel) [inline maps/interiors/apc_aa.dmm:23]: should have density = 1

**Review by AndroBetel — DISMISSED** [review]

File: code/modules/vehicles/hardpoints/primary/aa_cannon.dm (hunk 5 above)
Comment (BonniePandora) [reply code/modules/vehicles/hardpoints/primary/aa_cannon.dm:6]: Ah, alpha/transparency didn't transfer nicely across from the files Boom sent me.
I'll get that sorted

File: maps/interiors/apc_aa.dmm (hunk 9 above)
Comment (BonniePandora) [reply maps/interiors/apc_aa.dmm:23]: All requested changes addressed.

File: code/modules/vehicles/hardpoints/secondary/aa_launcher.dm (hunk 7 above)
Comment (BonniePandora) [reply code/modules/vehicles/hardpoints/secondary/aa_launcher.dm:32]: Upped the actual impact damage (given the ammo type is meant to be a hypersonic kinetic-kill kind of missile) and speed of the projectile massively, keeping the low mag-count to offset the fact this thing is meant to hit like a truck

**Pull request merged**